app/routes.py

```diff
@@ -1,2 +1,70 @@
-from flask import Blueprint
+from app import db
+from flask import Blueprint, jsonify, abort, make_response, request
+from .models.planet import Planet
 
+bp = Blueprint("planets", __name__, url_prefix="/planets")
+
+
+def validate_planet(planet_id):
+    try:
+        planet_id = int(planet_id)
+    except:
+        abort(make_response({"message": f"planet {planet_id} invalid"}, 400))
+
+    planet = Planet.query.get(planet_id)
+    if not planet:
+        abort(make_response({"message": f"planet {planet_id} not found"}, 404))
+
+    return planet
+
+def validate_planet_dict(request_body):
```

Comment:
Great idea to make a helper to check the values in the dictionary! I'm curious why you wouldn't allow creating a planet with a single ring?

If we spent more time with this code, we would want to think about how to separate some of the concerns here. For instance, part of why we moved the responsibility for how to create a record from a dictionary into the class itself was to relieve routes from needing to know what data a particular model required (the model already knows this). But here, we've brought some of that logic back into this validation method.

Teasing apart these dependencies can be tricky, and lead to code that is more complicated on the surface, but is usually done for the benfits of clarity and maintainability. But again, great idea!

Reply:
Oh, your rings is a boolean value. Be caureful, since in python `True` compares equal to `1`. If the intent was simply to require the presence of the key, prefer using an `in` check.